examples/platforms/efr32mg13/flash.c

```diff
@@ -28,22 +28,373 @@
+    static nvm3_Init_t name = {                                               \
+        (nvm3_HalPtr_t)name##_nvm,                                            \
+        sizeof(name##_nvm),                                                   \
+        name##_cache,                                                         \
+        sizeof(name##_cache) / sizeof(nvm3_CacheEntry_t),                     \
+        maxObjectSize,                                                        \
+        repackHeadroom,                                                       \
+        &nvm3_halFlashHandle,                                                 \
+    }
+
+#define NUM_SETTINGS_OBJECTS 7 // == number of enumerated Settings key types (in settings.hpp).
+#define NUM_INDEXED_SETTINGS \
+    OPENTHREAD_CONFIG_MLE_MAX_CHILDREN // Indexed key types are only supported for kKeyChildInfo (=='child table').
+#define NUM_USER_OBJECTS 16 // User nvm3 objects (nvm3 key range 0x0000 -> 0xDFFF is available for user data).
+
+#define OT_NVM3_FLASH_SIZE \
+    SETTINGS_CONFIG_PAGE_NUM *SETTINGS_CONFIG_PAGE_SIZE // == 4 x (Flash page size:MG12/MG13=2K, MG21=8K)
+#define OT_NVM3_MAX_NUM_OBJECTS NUM_SETTINGS_OBJECTS + NUM_INDEXED_SETTINGS + NUM_USER_OBJECTS
+#define OT_NVM3_MAX_OBJECT_SIZE 256
+#define OT_NVM3_REPACK_HEADROOM 64 // Threshold for automatic nvm3 flash repacking.
+
+#define OT_NVM3_SETTINGS_KEY_PREFIX 0xE000 // (=> key range 0x0000-0xDFFF is available for OT User nvm3 objects).
+
+#define ENUM_NVM3_KEY_LIST_SIZE 4 // List size used when enumerating nvm3 keys.
+
+static nvm3_Handle_t handle = {NULL};
+
+static otError          addSetting(uint16_t aKey, const uint8_t *aValue, uint16_t aValueLength);
+static nvm3_ObjectKey_t makeNvm3ObjKey(uint16_t otSettingsKey, int index);
+static otError          mapNvm3Error(Ecode_t nvm3Res);
+
+// Declare OT NVM3 data area and cache.
+// OT data area should be enough for OT 'Settings' and whatwever User data is required.
+
+OT_NVM3_DEFINE_SECTION_STATIC_DATA(otNvm3, OT_NVM3_FLASH_SIZE, OT_NVM3_MAX_NUM_OBJECTS);
+
+OT_NVM3_DEFINE_SECTION_INIT_DATA(otNvm3, OT_NVM3_MAX_OBJECT_SIZE, OT_NVM3_REPACK_HEADROOM);
+
+void otPlatSettingsInit(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    otError err;
+    bool    needClose = false;
+
+    err = mapNvm3Error(nvm3_open(&handle, &otNvm3));
+    SuccessOrExit(err);
+    needClose = true;
+
+exit:
+    if (needClose)
+    {
+        nvm3_close(&handle);
+    }
+}
+
+void otPlatSettingsDeinit(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    nvm3_close(&handle);
+}
+
+otError otPlatSettingsGet(otInstance *aInstance, uint16_t aKey, int aIndex, uint8_t *aValue, uint16_t *aValueLength)
+{
+    // Searches through all matching nvm3 keys to find the one with the required
+    // 'index', then reads the nvm3 data into the destination buffer.
+    // (Repeatedly enumerates a list of matching keys from the nvm3 until the
+    // required index is found).
```

Comment:
Do we need to add a comment to define the behaviour of aLevel

Reply:
Comment could be expanded when the efr32 platform code is made common.